mlflow/langchain/databricks_dependencies.py

```diff
@@ -48,6 +48,23 @@ def _extract_databricks_dependencies_from_retriever(retriever) -> Generator[Reso
             yield DatabricksServingEndpoint(endpoint_name=embeddings.endpoint)
 
 
+def _extract_databricks_dependencies_from_retriever(retriever) -> Generator[Resource, None, None]:
+    if hasattr(retriever, "base_retriever"):
+        retriever = getattr(retriever, "base_retriever", None)
+
+    if hasattr(retriever, "retriever"):
+        retriever = getattr(retriever, "retriever", None)
+
+    if hasattr(retriever, "retrievers"):
+        retriever = getattr(retriever, "retrievers", None)
```

Comment:
non blocking but curious why these are needed?

Reply:
+1, could you add comments or docstring explaining for which case we need this extracting? While we can find some clues in LangChain code base, as their code base changes rapidly, it is often helpful to have some note in our code base🙂